### ISSUE TITLE
Regarding Testing

### ISSUE THREAD

**Issue by jeromkiller** [jeromkiller/GBemu#4]
During the implementation of the opcodes i have been second guessing myself regarding the implementation of some opcodes.
especialy when it comes to the carry and half carry flags.

I'd really like to be able to test the emulator through blargg's test suite (https://github.com/retrio/gb-test-roms).

I don't have to implement user input or implement the LCD because all output to screen is also written to the link port.
To implement the test suite i have opened a project (https://github.com/jeromkiller/GBemu/projects/2)

after which i should be able to validate if the implementation of the opcodes is correct

**Comment by jeromkiller** [comment]
The basic implementation of the opcodes is done,
next i think i'm going to add the rom mapper for which i will have to do a bit more stirring in the LD8 opcode

**Comment by jeromkiller** [comment]
the basic rom mapper is in place, and basic logic seems to work fine for running the test rom so far.
In the cpu_intsrs.gb file there is an issue where it seems to output tests that don't exits.
I think that might be an issue in the rom mapper regarding bank switching, which i haven't tested the implementation of yet.

I have also added a new main file for running tests which will probably stay more bare bones (no display, no input, no sound, not slowed) to keep tests quick.

the makefile has also been updated to build the test version, and to be able to run a script and clean up afterwards.

i guess i'll have to write a little script that runs each test and reports on the results
